src/ImageSharp/Formats/Jpeg/Components/FastFloatingPointDCT.cs

```diff
@@ -309,11 +198,106 @@ public static void FDCT8x4_RightPart(ref Block8x8F s, ref Block8x8F d)
         }
 
         /// <summary>
-        /// Apply floating point IDCT transformation into dest, using a temporary block 'temp' provided by the caller (optimization)
+        /// Combined operation of <see cref="FDCT8x4_LeftPart(ref Block8x8F, ref Block8x8F)"/> and <see cref="FDCT8x4_RightPart(ref Block8x8F, ref Block8x8F)"/>
+        /// using AVX commands.
+        /// </summary>
+        /// <param name="s">Source</param>
+        /// <param name="d">Destination</param>
+        private static void FDCT8x8_Avx(ref Block8x8F s, ref Block8x8F d)
+        {
+#if SUPPORTS_RUNTIME_INTRINSICS
+            Debug.Assert(Avx.IsSupported, "AVX is required to execute this method");
+
+            Vector256<float> t0 = Avx.Add(s.V0, s.V7);
+            Vector256<float> t7 = Avx.Subtract(s.V0, s.V7);
+            Vector256<float> t1 = Avx.Add(s.V1, s.V6);
+            Vector256<float> t6 = Avx.Subtract(s.V1, s.V6);
+            Vector256<float> t2 = Avx.Add(s.V2, s.V5);
+            Vector256<float> t5 = Avx.Subtract(s.V2, s.V5);
+            Vector256<float> t3 = Avx.Add(s.V3, s.V4);
+            Vector256<float> t4 = Avx.Subtract(s.V3, s.V4);
+
+            Vector256<float> c0 = Avx.Add(t0, t3);
+            Vector256<float> c1 = Avx.Add(t1, t2);
+
+            // 0 4
+            d.V0 = Avx.Add(c0, c1);
+            d.V4 = Avx.Subtract(c0, c1);
+
+            Vector256<float> c3 = Avx.Subtract(t0, t3);
+            Vector256<float> c2 = Avx.Subtract(t1, t2);
+
+            // 2 6
+            d.V2 = SimdUtils.HwIntrinsics.MultiplyAdd(Avx.Multiply(c2, C_V_0_5411), c3, C_V_1_3065);
+            if (Fma.IsSupported)
+            {
+                d.V6 = Fma.MultiplySubtract(c3, C_V_0_5411, Avx.Multiply(c2, C_V_1_3065));
+            }
+            else
+            {
+                d.V6 = Avx.Subtract(Avx.Multiply(c3, C_V_0_5411), Avx.Multiply(c2, C_V_1_3065));
+            }
```

Comment:
I would define a method `SimdUtils.HwIntrinsics.MultiplySubtract` to get rid of the if blocks.

Reply:
Forgot about this, will do!